File: programacion-orientada-a-objetos/CursoPOOUber/Python/car.py
from account import Account

class Car:
    id          = int
    license     = str
    driver      = Account("","")
    passenger   = int

    def __init__(self, license, driver):
        self.license    = license
        self.driver     = driver
    
    def printDataCar(self, uber): #Todas las funciones deben llevar "self"
        if self.passenger is not None:
            print(vars(uber))

    def setPassenger(self, passenger):
        if passenger == 4:
            self.passenger = passenger
        else:
            self.passenger = None
            print("Debe asignar 4 pasajeros")

File: programacion-orientada-a-objetos/CursoPOOUber/PHP/Car.php
<?php
require_once('account.php');
class Car {
    public $id;
    public $license;
    public $driver;
    protected $passenger;
    //Constructor
    public function __construct($license, $driver) {
        $this->license = $license;
        $this->driver = $driver;
    }

    //General functions
    public function printDataCar() {
        echo "
            Licencia: $this->license 
            Driver: {$this->driver->name} 
            Número de pasajeros: $this->passenger\n
        ";
    }


    //Getters and Setters
    public function getPassenger(){
        return $this->passenger;
    }
    public function setPassenger($passenger){
        if($passenger == 4){
            $this->passenger = $passenger;
        }
        else{
            echo "Necesitas asignar 4 pasajeros";
        }
    }
}
?>